program-interface/Cargo.toml

```diff
@@ -0,0 +1,26 @@
+[package]
+name = "solana-program-interface"
```

Comment:
I'm confused about the name and scope of this crate.  That `-interface` suffix suggests it'll only be a dependency of `instruction` and `state` modules, and that `processor` modules might require an additional dependency. Is that correct? If so, why is `entrypoint.rs` here? If no, can we rename to `solana-program`?